docs/Contributing/Advanced-Configuration.md

```diff
@@ -0,0 +1,378 @@
+# Advanced Configuration
+
+TODO(noah): some context to explain those non-recommended settings.
```

Comment:
The new page in "Contributing" to document the settings that are available but unrecommended to use in the config YAML (due to requiring secrets/password).